colink/application.py

```diff
@@ -36,20 +39,43 @@ def __init__(
 
 class CoLinkSubscriber:
     def __init__(self, mq_uri: str, queue_name: str):
-        self.uri = mq_uri
+        uri_parsed = urlparse(mq_uri)
         self.queue_name = queue_name
-        param = pika.connection.URLParameters(url=self.uri)
-        mq = pika.BlockingConnection(param)  # establish rabbitmq connection
-        self.channel = mq.channel()
+        if uri_parsed.scheme.startswith("redis"):
+            r = redis.from_url(mq_uri)
+            self.mq_type = "redis"
```

Comment:
Just a note that we can actually use a polymorphism + factory pattern here to simplify a bit (however in Rust it is not recommended), but consider we only have two different mq types, maybe it's also fine to keep it as is.
E.g., if we are separating CoLinkRedisSubscriber and CoLinkRabbitMQSubscriber, we don't have to maintain the useless self.rabbitmq_channel in the Redis instance, and redis_connection in the RabbitMQ instance.